### PR TITLE
Migrate scalar displays

### LINKED ISSUE #90: Migrate TemperatureDisplay

### PR DESCRIPTION
**Pull request by GW1708**
Closes #90 #89 #78 #77 

This PR migrates temperature, illuminance, relative humidity and fluid pressure displays.
Since they work extremely similar they were refactored by introducing a point cloud scalar parent class

Refactored display page objects by moving set queue size to point cloud common page object and deleting unnecessary page objects.

CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5934/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2460/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4904/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5823/)

Including Visual Tests:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5933/)

### CODE REVIEW

**Comment by GW1708** [comment]
CI was done by @Martin-Idel-SI 
Visual tests on Linux seem to fail but are fine locally and on our Jenkins

File: rviz_default_plugins/CMakeLists.txt [hunk 1]
```diff
@@ -190,7 +199,7 @@ set(rviz_default_plugins_source_files
   src/rviz_default_plugins/view_controllers/orbit/orbit_view_controller.cpp
   src/rviz_default_plugins/view_controllers/ortho/fixed_orientation_ortho_view_controller.cpp
   src/rviz_default_plugins/view_controllers/xy_orbit/xy_orbit_view_controller.cpp
-)
+        test/rviz_default_plugins/publishers/fluid_pressure_publisher.hpp)
```
Comment (wjwwood) [inline rviz_default_plugins/CMakeLists.txt]:
```suggestion
  test/rviz_default_plugins/publishers/fluid_pressure_publisher.hpp
)
```

Comment (wjwwood) [reply rviz_default_plugins/CMakeLists.txt]: Why is this file added to the sources list? Shouldn't it instead be part of the test executable?

File: rviz_default_plugins/CMakeLists.txt [hunk 2]
```diff
@@ -480,6 +489,15 @@ if(BUILD_TESTING)
     target_link_libraries(point_cloud_common_test ${TEST_LINK_LIBRARIES})
   endif()
 
+  ament_add_gmock(point_cloud_scalar_display_test
+          test/rviz_default_plugins/displays/pointcloud/point_cloud_scalar_display_test.cpp
+          test/rviz_default_plugins/displays/display_test_fixture.cpp
+          ${SKIP_DISPLAY_TESTS})
```
Comment (wjwwood) [inline rviz_default_plugins/CMakeLists.txt]: Please use 2 space indentations.

Comment (wjwwood) [reply rviz_default_plugins/CMakeLists.txt]: Here and below.

File: rviz_default_plugins/include/rviz_default_plugins/displays/fluid_pressure/fluid_pressure_display.hpp [hunk 3]
```diff
@@ -27,51 +28,43 @@
...
-
-  virtual void update( float wall_dt, float ros_dt );
-
-private Q_SLOTS:
-  void updateQueueSize();
+/**
+* \class FluidPressureDisplay
+* \brief Displays an FluidPressure message of type sensor_msgs::FluidPressure
+*
+*/
```
Comment (wjwwood) [inline rviz_default_plugins/include/rviz_default_plugins/displays/fluid_pressure/fluid_pressure_display.hpp]: This should be like this:

```
/// Display a FluidPressure message of type sensor_msgs::FluidPressure.
/**
 * \class FluidPressureDisplay
 */
```

Note: that the `*`'s align vertically, `Display` and not `Displays`, and `a` not `an`.

File: rviz_default_plugins/include/rviz_default_plugins/displays/fluid_pressure/fluid_pressure_display.hpp [hunk 4]
```diff
@@ -27,51 +28,43 @@
...
-  virtual void reset();
-
-  virtual void update( float wall_dt, float ros_dt );
-
-private Q_SLOTS:
-  void updateQueueSize();
+/**
+* \class FluidPressureDisplay
+* \brief Displays an FluidPressure message of type sensor_msgs::FluidPressure
+*
```
Comment (wjwwood) [inline rviz_default_plugins/include/rviz_default_plugins/displays/fluid_pressure/fluid_pressure_display.hpp]:
```suggestion
```

File: rviz_default_plugins/include/rviz_default_plugins/displays/fluid_pressure/fluid_pressure_display.hpp [hunk 5]
```diff
@@ -27,51 +28,43 @@
...
 
-  virtual void reset();
-
-  virtual void update( float wall_dt, float ros_dt );
-
-private Q_SLOTS:
-  void updateQueueSize();
+/**
+* \class FluidPressureDisplay
+* \brief Displays an FluidPressure message of type sensor_msgs::FluidPressure
```
Comment (wjwwood) [inline rviz_default_plugins/include/rviz_default_plugins/displays/fluid_pressure/fluid_pressure_display.hpp]: Move the brief above the `/**` block comment and use `///`.

File: rviz_default_plugins/include/rviz_default_plugins/displays/fluid_pressure/fluid_pressure_display.hpp (hunk 3 above)
Comment (wjwwood) [reply rviz_default_plugins/include/rviz_default_plugins/displays/fluid_pressure/fluid_pressure_display.hpp]: These doc style changes should be applied to the rest of the doc blocks in the rest of this file.

File: rviz_default_plugins/plugins_description.xml [hunk 6]
```diff
@@ -121,6 +143,16 @@
     <message_type>sensor_msgs/Range</message_type>
   </class>
 
+  <class
+          name="rviz_default_plugins/RelativeHumidity"
```
Comment (wjwwood) [inline rviz_default_plugins/plugins_description.xml]: 2 space indentation here too.

**Review by wjwwood — APPROVED** [review]
lgtm, just with a few style nitpick's. I'll run CI and build/test locally asap.

File: rviz_default_plugins/CMakeLists.txt (hunk 1 above)
Comment (GW1708) [reply rviz_default_plugins/CMakeLists.txt]: My bad, it should be part of the test executable of course. Fixed!

**Comment by GW1708** [comment]
Done! Fixed the remaining style errors and added a missing keyword to the visual testing framework README

**Comment by andreasholzner** [comment]
Just to be sure a new round of CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5982/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2496/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4939/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5862/)

Visual tests: [Build Status](https://ci.ros2.org/job/ci_linux/5981/)

**Comment by andreasholzner** [comment]
@wjwwood The unstable builds of the previous round of CI were all uncrustify errors that are already fixed by #366. So I rebased and started new CI build just to be sure (we no longer have our own CI infrastructure at TNG available). The Linux failure above was repeatedly caused by a docker connection error and is not related to the code problems.
 
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5983/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2497/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4940/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5863/)

**Comment by wjwwood** [comment]
Trying Linux again:

- [Build Status](https://ci.ros2.org/job/ci_linux/5989/)

**Review by wjwwood — APPROVED** [review]

**Pull request merged**